### ISSUE TITLE
Try to do Validation while Training

### ISSUE THREAD

**Issue by bzhong2** [kwotsin/transfer_learning_tutorial#4]
Hello, 
It will be beneficial to evaluate the model with validation set while training so that we can get a better ideas whether the model got overfit. I modified the code as below. I reused all the variables while only changing the input to be validation images. I then did session.run(accuracy_train,accuracy_validation) in the training step function. However, in the output the validation accuracy is always 0 while training accuracy is alright: 
INFO: tensorflow:global step 18: loss: 1.1491 Train accuracy :0.5481  Validation accuracy :0.0000(24.39 sec/step)

I might have chosen a stupid way, is there any simpler or better way to do it? I was thinking about using validation monitor but there is no good example how to integrate it into this code. Any help is appreciated and the following is how I modified the code.  

predictions = tf.argmax(end_points['Predictions'], 1)
probabilities = end_points['Predictions']
accuracy, accuracy_update = tf.contrib.metrics.streaming_accuracy(predictions, labels)
metrics_op = tf.group(accuracy_update, probabilities)
tf.get_variable_scope().reuse_variables()
with slim.arg_scope(inception_arg_scope()):	
        logits_validation, end_points_validation = inception_v3(raw_images_validation, num_classes = 
                                                                dataset_validation.num_classes, reuse = True, is_training = False)
predictions_validation = tf.argmax(end_points_validation['Predictions'], 1)
probabilities_validation = end_points_validation['Predictions']
accuracy_validation, accuracy_update_validation = 
                                           tf.contrib.metrics.streaming_accuracy(predictions_validation, labels_validation)

In training step function : 
total_loss, global_step_count, accuracy_value_train,accuracy_value_validation,_ = sess.run([train_op, global_step, accuracy, accuracy_validation, metrics_op])

**Comment by kwotsin** [comment]
Yes indeed. That was an issue I had when I first wrote the code, as I found it might take too much memory to construct two graphs with `is_training=False` and another one with `is_training=True` to do the validation. I think it's possible to do some compromise by letting `is_training=True` to be used for validation images, but then again you would need to have unseen images, which would require you to modify the data pipeline a little to get train, validation and test data instead of simply just the train and test data currently. Even as you do so, you would have to make sure your model doesn't train on the validation/test data, so I would guess the code will become rather complicated or too bulky to be used for training. I have not found a workaround for this unfortunately..

Currently what I do is I run the evaluation code periodically to see whether the data has overfit, but this is a manual method. You can also try writing a script to automate this process.
